storage/local/series.go

```diff
@@ -175,28 +178,29 @@ type memorySeries struct {
 }
 
 // newMemorySeries returns a pointer to a newly allocated memorySeries for the
-// given metric. reallyNew defines if the memorySeries is a genuinely new series
-// or (if false) a series for a metric being unarchived, i.e. a series that
-// existed before but has been evicted from memory. If reallyNew is false,
-// firstTime is ignored (and set to the lowest possible timestamp instead - it
-// will be set properly upon the first eviction of chunkDescs).
-func newMemorySeries(
-	m clientmodel.Metric,
-	reallyNew bool,
-	firstTime clientmodel.Timestamp,
-) *memorySeries {
-	if !reallyNew {
-		firstTime = clientmodel.Earliest
-	}
-	s := memorySeries{
-		metric:          m,
-		headChunkClosed: !reallyNew,
-		savedFirstTime:  firstTime,
-	}
-	if !reallyNew {
-		s.chunkDescsOffset = -1
-	}
-	return &s
+// given metric. chunkDescs and modTime in the new series are set according to
+// the provided parameters. chunkDescs can be nil or empty if this is a
+// genuinely new time series (i.e. not one that is being unarchived). In that
+// case, headChunkClosed is set to false, and firstTime and lastTime are both
+// set to clientmodel.Earliest. The zero value for modTime can be used if the
+// modification time of the series file is unknown (e.g. if this is a genuinely
+// new series).
+func newMemorySeries(m clientmodel.Metric, chunkDescs []*chunkDesc, modTime time.Time) *memorySeries {
```

Comment:
I find it very confusing to suddenly switch to native `time.Time` after using `clientmodel.Timestamp` everywhere else. The `modTime` field has been `time.Time` all along – is this due to a need for sub-millisecond precision?